### DISCUSSION TITLE
Settings for Apple/iCloud Accounts

### DISCUSSION THREAD

**Discussion by johnt-tally** [moralmunky/Home-Assistant-Mail-And-Packages#658] [Q&A]
Does anyone have this working with your Apple email account?  Because I have 2FA enabled, I created an app specific password for this specific integration.  But I don't think it is working based on some repeating error messages in HA:  This error originated from a custom integration.

Logger: custom_components.mail_and_packages
Source: helpers/update_coordinator.py:196
Integration: Mail and Packages (documentation, issues)
First occurred: 12:59:10 PM (9 occurrences)
Last logged: 1:02:31 PM

Timeout fetching Mail and Packages (imap.mail.me.com) data

Any ideas?

**Comment by johnt-tally** [comment]
Update--I'm on the latest non-beta version (0.3.3-2) running on 2022.4.5 for Home Assistant.  The repeated error message now are: 2022-04-18 09:04:43 ERROR (MainThread) [custom_components.mail_and_packages] Error fetching Mail and Packages (imap.mail.me.com) data: 'NoneType' object has no attribute 'split'
2022-04-18 09:04:43 ERROR (MainThread) [custom_components.mail_and_packages] Error updating sensor data: 'NoneType' object has no attribute 'split'

**Reply by firstof9** [reply]
Try the most recent beta, but odds are low that Apple mail will work.

**Reply by krbuck** [reply]
I had similar issues around the app password and actually saw also the errors you are getting. I switched back to my old gmail account to resolve this, but would love to help make my icloud account workable.

**Reply by DolfLund** [reply]
I have gone over and over with this setup, I'm stuck with icloud (using app password, which I know works because my folder list is accurate). 

Error I see is : DEBUG email_search value was invalid: None
and
DEBUG email_search value was invalid: None

**Reply by firstof9** [reply]
Correct, those are non-standard replies from the mail server.

**Reply by johnt-tally** [reply]
And besides trying the latest beta, and other ideas?  (And thanks for your feedback!)

**Reply by firstof9** [reply]
I guess you could complain to apple to follow IMAP RFC?

Specifically: https://datatracker.ietf.org/doc/html/rfc3501#section-6.4.4

Their search function is returning "None" rather than just nothing.
Example:

Apple:
               C: A283 SEARCH TEXT "string not in mailbox"
               S: * SEARCH None

Proper reply:
               C: A283 SEARCH TEXT "string not in mailbox"
               S: * SEARCH

**Comment by asinla** [comment]
Did anyone ever get this to work with Apple/iCloud accounts?

**Comment by taw123** [comment]
Not yet. In the same boat here trying to get this to work

**Comment by asinla** [comment]
Any updates on getting Apple mail to work?

Thanks!

**Reply by firstof9** [reply]
Do you get errors when you try to use Apple mail?

**Reply by asinla** [reply]
Yup. Errors.

**Reply by firstof9** [reply]
Is it this error?
`Error updating sensor data: 'NoneType' object has no attribute 'split'`

**Comment by b2dmx** [comment]
Does this really still not work for iCloud? They should put something in the docs…

**Reply by firstof9** [reply]
A fix for Apple's IMAP server has been implemented already.

**Reply by b2dmx** [reply]
> A fix for Apple's IMAP server has been implemented already.

Glad to hear it. How does one set this up? No guidance for Apple in the docs. 
What is the host? Does it need an app password or standard password?

**Reply by firstof9** [reply]
Looks like the answers are here: https://support.apple.com/en-us/102525

**Reply by b2dmx** [reply]
Thank you!
